### PR TITLE
Clarify how exceptions are caught in request#add_review

### PR DESCRIPTION
**Pull request by eduardoj**
Also move searching for the request to the beginning of the method, to fail fast and first in case that the request is not found.

See this thread of comments: https://github.com/openSUSE/open-build-service/pull/12844#discussion_r926929787

### CODE REVIEW

**Comment by dmarcoux** [comment]
Could we rephrase the commit message to `Clarify how exceptions are caught in request#add_review`?

**Comment by eduardoj** [comment]
> Could we rephrase the commit message to `Clarify how exceptions are caught in request#add_review`?

Sure. Done.

File: src/api/app/controllers/webui/request_controller.rb [hunk 1]
```diff
@@ -41,25 +41,17 @@ def create
...
-        opts[:by_project] = params[:review_project]
-        opts[:by_package] = params[:review_package]
-      end
-      opts[:comment] = params[:review_comment] if params[:review_comment]
-
       request = BsRequest.find_by_number!(params[:number])
-      request.addreview(opts)
+    rescue ActiveRecord::RecordNotFound => e
+      flash[:error] = "Unable to add review to request with id '#{params[:number]}': #{e.message}"
+      return
```
Comment (dmarcoux) [inline src/api/app/controllers/webui/request_controller.rb:47]: If somehow `ActiveRecord::RecordNotFound` is raised and caught in this `rescue` block, this `return` will cause the UI to simply not respond. This is quite confusing, see here:

nothing-happening

The solution is either to duplicate the `redirect_to` call on line 57 below or to reorganize the code.

Comment (eduardoj) [reply src/api/app/controllers/webui/request_controller.rb:47]: Ops, true. On it.

Comment (eduardoj) [reply src/api/app/controllers/webui/request_controller.rb]: Given that we are rescuing only `ActiveRecord::RecordNotFound` exceptions, I changed the code from a `find_by_number!` to a `find_by_number`, and later checked with `.nil?`.

**Review by dmarcoux — APPROVED** [review]

**Pull request merged**